libkbfs/block_journal.go

```diff
@@ -314,16 +358,16 @@ func (j *blockJournal) putData(
 		return err
 	}
 
-	err = j.s.put(id, context, buf, serverHalf, next.String())
+	putData, err := j.s.put(id, context, buf, serverHalf, next.String())
 	if err != nil {
 		return err
 	}
 
-	// Decremented when the put journal entry is ignored or
-	// flushed.
-	err = j.adjustUnflushedBytes(int64(len(buf)))
-	if err != nil {
-		return err
+	if putData {
```

Comment:
Dedup would be a put + addReference, right? I think this fixes a different bug, if we somehow had a case where we'd retry a put for the same block.